### PR TITLE
Add PR build action and git commit dockerfiles on changes

### PR DESCRIPTION
**Pull request by redhatrises**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    steps:
+
+    - name: Set up Go 
+      uses: actions/setup-go@v2
+      id: go
+
+    - name: Check out Dockerfile template code into the directory
+      uses: actions/checkout@v2
+      with:
+        repository: "redhatrises/dockerfile-template"
```
Comment (shawndwells) [inline .github/workflows/build.yml:26]:
```suggestion
        repository: "redhatrises/dockerfile-template"
```

Comment (shawndwells) [inline .github/workflows/build.yml:26]: your personal repo?

Comment (redhatrises) [reply .github/workflows/build.yml:26]: Currently, yes. We have several of these small things not under the CS org as they function like github-actions, and ultimately, might be removed in the long run when there is full support.

Comment (redhatrises) [reply .github/workflows/build.yml:26]: Plus was already reviewed in PR #34

Comment (isimluk) [reply .github/workflows/build.yml:26]: I am Ok, with using personal repo's from time to time. In my mind, it is fine during the initial phases, when you have something and you are not 100% clear about the future of that something. Also, I am ok with personal repos in the additional CI pipelines.

**Review by isimluk — APPROVED** [review]
👍 I am eager to see if this works.

@redhatrises, would you mind creating small pr to the template to verify this approach?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+name: Build Dockerfiles from template
+
+env:
+  GO111MODULE: on
+
+on:
+  pull_request:
+    branches: [ main ]
+  push:
+    branches: [ main ]
```
Comment (isimluk) [inline .github/workflows/build.yml:10]: @redhatrises WDYT about removing this line. It should force the pipeline run on PRs and thus allow users to see what they have done.

**Pull request merged**